src/Algebra/Properties/Semiring/Primality.agda

```diff
@@ -1,22 +1,31 @@
 ------------------------------------------------------------------------
 -- The Agda standard library
 --
--- Some theory for CancellativeCommutativeSemiring.
+-- Properties of Coprimality and Irreducibility for Semiring.
 ------------------------------------------------------------------------
 
 {-# OPTIONS --cubical-compatible --safe #-}
 
-open import Algebra using (Semiring)
-open import Data.Sum.Base using (reduce)
-open import Function.Base using (flip)
-open import Relation.Binary.Definitions using (Symmetric)
+open import Algebra.Bundles using (Semiring)
 
 module Algebra.Properties.Semiring.Primality
   {a ℓ} (R : Semiring a ℓ)
   where
 
-open Semiring R renaming (Carrier to A)
+open import Data.Sum.Base using (reduce)
+open import Function.Base using (flip)
+open import Relation.Binary.Definitions using (Symmetric)
+
+open Semiring R
+  using (_≉_; sym; trans; 0#; 1#; zeroˡ; rawSemiring)
```

Comment:
Is this refinement of `using` worth the extra two lines? We're in `Properties.Semiring` so if we have name clashes with the remaining fields in`Semiring` I feel that its probably a bug and it would be better to detect it.

Reply:
Yes, I agree. Will revert!